### PR TITLE
feat(aibom): improve AI component detection for LangGraph, CrewAI, and LangChain

### PR DESCRIPTION
**Pull request by vineethsai7**
## Summary

- **Supplemental catalog**: Add 80+ in-memory entries for LangGraph, CrewAI, and LangChain symbols missing from the prebuilt DuckDB catalog (StateGraph, Agent, Flow, OpenAIEmbeddings, InMemorySaver, etc.)
- **Notebook + config parsing**: Support Jupyter `.ipynb` files and `langgraph.json` config files as new input sources for the analyzer
- **Broader detection**: Detect standalone function calls (not just assignments/decorators), resolve method chains (e.g. `builder.compile()` → `StateGraph.compile`), and derive `USES_MEMORY`, `USES_RETRIEVER`, `USES_EMBEDDING` relationships in addition to existing `USES_LLM` and `USES_TOOL`

### New modules
| File | Purpose |
|------|---------|
| `supplemental_catalog.py` | In-memory catalog entries for LangGraph, CrewAI, LangChain symbols not in DuckDB |
| `notebook_parser.py` | Extract Python code from `.ipynb` files, handle IPython magics |
| `config_parser.py` | Parse `langgraph.json` for graph entrypoints and embedding metadata |

### Modified modules
| File | Change |
|------|--------|
| `catalog_db.py` | Query supplemental catalog alongside DuckDB; DuckDB entries take precedence |
| `categorizer.py` | Process standalone calls; add memory/retriever/embedding relationships; method-chain resolution via variable_map |
| `cst_parser.py` | Fallback to `_format_attribute_name` when qualified name resolution fails for attribute-access calls |
| `cli.py` | Discover `.ipynb` files; invoke notebook and config parsers in analysis loop |
| `.gitignore` | Ignore `aibom/benchmarks/_repos/` (cloned benchmark repos) |

### Tests
- `test_categorizer.py`: 4 new cases (standalone calls, USES_MEMORY, USES_RETRIEVER, USES_EMBEDDING)
- `test_notebook_parser.py`: new file covering extraction + edge cases
- `test_config_parser.py`: new file covering langgraph.json parsing
- `test_supplemental_catalog.py`: new file validating entry structure and uniqueness

## Benchmark results (not included in PR)

Ran against 13 real-world repos from langchain-ai and crewAIInc (1,560 files, 6,930 components detected):
- **Detection precision ~95%** (almost all detected items are real AI symbols)
- **Category accuracy ~56%** (dragged down by DuckDB mapping `crewai.Crew`/`crewai.Task` to `concept="model"`)
- Top FP patterns: `typing.cast`/`typing.Annotated` (stdlib), `crewai.Crew.kickoff` (execution call)
- Top FN patterns: `langchain_tavily.TavilySearch`, `langgraph.prebuilt.ToolNode` (not yet in catalog)

## Test plan

- [x] All existing unit tests pass (`uv run pytest`)
- [x] New unit tests cover standalone calls, new relationship types, notebook parsing, config parsing, and supplemental catalog
- [x] Benchmark runner validated against 13 repos
- [ ] Manual review of category accuracy for crewAI symbols (known DuckDB issue documented)

### CODE REVIEW

File: aibom/src/aibom/config_parser.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Copyright 2026 Cisco Systems, Inc. and its affiliates
```
Comment (skamanbo) [inline aibom/src/aibom/config_parser.py:1]: change file name to supplemental_catalog_parser.py?

File: aibom/src/aibom/config_parser.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+def parse_langgraph_json(project_root: Path) -> Optional[CodeAnalysisResult]:
+    """Parse ``langgraph.json`` at *project_root* and emit synthetic observations.
+
+    Extracts:
+    - ``graphs`` entries as agent entrypoints
+    - ``store.index.embed`` as an embedding model component
+    - ``store.index.dims`` as metadata on the embedding
+    """
+    config_path = project_root / "langgraph.json"
```
Comment (skamanbo) [inline aibom/src/aibom/config_parser.py:44]: I see SUPPLEMENTAL_ENTRIES, do we still need lanograph.json ?

Comment (vineethsai7) [reply aibom/src/aibom/config_parser.py:44]: I added this PR hoping it to be modified, so you can add these to the duckdb. We should include all of this directly in duckdb and only cherry pick from this PR.

File: aibom/src/aibom/config_parser.py (hunk 1 above)
Comment (vineethsai7) [reply aibom/src/aibom/config_parser.py:1]: Addressed in this branch: renamed to `framework_config_parser.py` instead of `supplemental_catalog_parser.py`. The supplemental catalog module has been deleted entirely -- all its entries are now in DuckDB. The remaining parser handles `langgraph.json` framework config files, so `framework_config_parser` is more accurate.

File: aibom/src/aibom/config_parser.py (hunk 2 above)
Comment (vineethsai7) [reply aibom/src/aibom/config_parser.py:44]: Yes, `langgraph.json` parsing is still needed. It extracts runtime configuration that isn't visible in Python source analysis: graph entrypoints (`graphs.agent = ./src/graph.py:graph`) and embedding models (`store.index.embed = openai:text-embedding-3-small`). The supplemental *catalog entries* (SUPPLEMENTAL_ENTRIES) were the redundant part -- those 105+ entries are now built into DuckDB via `scripts/build_catalog.py`. The parser itself serves a different purpose.

**Comment by vineethsai7** [comment]
Superseded by comprehensive PR that consolidates all catalog improvements, bug fixes, and new framework coverage into a single changeset.
